lib/mvi/responses/base.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+module MVI
+  module Responses
+    class Base
+      attr_accessor :code, :query, :original_response
+
+      RESPONSE_CODES = {
+        success: 'AA',
+        failure: 'AE',
+        invalid_request: 'AR'
+      }.freeze
+
+      class << self
+        attr_accessor :endpoint
+      end
+
+      def self.mvi_endpoint(endpoint)
+        @endpoint = endpoint
+      end
+      delegate :endpoint, to: 'self.class'
```

Comment:
I worry this could be a 'gotcha' for someone handling an instance of a class that extends `Base`.  For example,
```ruby
Foo.new.endpoint = 'my_endpoint'
```
without looking at `MVI::Responses:Base`, I would automatically assume `endpoint` to be an instance variable.